File: BestTimeToBuyAndSellStock2.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
    
public:
    
    int maxProfit(vector<int>& prices) {
        int n=prices.size();
        int dp[n+1];
        int i,j,k;
        for(i=0;i<=n;i++)
        {
            dp[i]=0;
        }
        for(i=1;i<=n;i++)
        {
           
            for(j=i;j<=n;j++)
            {
                dp[j]=max(dp[i]+prices[j-1]-prices[i-1],dp[j-1]);
               
            }
            
        }
       
        return dp[n];
    }
};
int main()
{
    
    return 0;
}

File: BestTimeToBuyAndSellStock2.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let profit=0;
    let n=prices.length;
    for(let i=1;i<n;i++)
    {
        profit+=Math.max(0,prices[i]-prices[i-1]);
    }
    return profit;
};